lib/haskell/natural4/src/LS/XPile/MathLang/GenericMathLang/ToGenericMathLang.hs

```diff
@@ -83,8 +82,14 @@ toMathLangGen l4i =
 
 -- Utility functions for expanding rules and inserting TypeDecls into GIVENs
 -- (Introduced in 2024-06, I hope we deal with global vs. local variables better later.)
-getHornlikes l4i = l4i.origrules ^.. folded % cosmosOf (gplate @Rule) % filteredBy (_Ctor @"Hornlike")
-getTypeDecls l4i = l4i.origrules ^.. folded % cosmosOf (gplate @Rule) % filteredBy (_Ctor @"TypeDecl")
+
+-- | Extract all 'Hornlike' rules.
+getHornlikes :: Interpreted -> [Rule]
+getHornlikes l4i = l4i.origrules ^.. folded % cosmosOf (gplate @Rule) % filteredBy _Hornlike
```

Comment:
Thanks for this too --- I'll try to remember to use the `_Hornlike` prisms instead of `_Ctor ...` next time!

Reply:
I'm actually not sure if the whole function is correct. What is the intent here? Should all hornlikes be extracted from all `Rule`s, even sub-rules? Or should only top-level hornlikes be extracted?

If the former, then what's the justification for just dropping everything around such deeply nested hornlikes.

If the latter, then it should probably be a simple `filter` call overall, nothing more.

Reply:
Good catch. my version of 'getHornlikes' had just been a wordier version of filter (it was `l4a.origrules ^.. folded % filteredBy (_Ctor @"Hornlike")`); I'm guessing actually that a filter is all we want, but will have to check with @inariksit .

(And similarly with `getVarDecls` -- though here I suspect Inari probably had a reason for wanting to use `cosmos`; I just can't remember what now)

Reply:
I think the reason why I added cosmos was to get access to potential Hornlikes in the `where` fields of the top level rules. But in retrospect, I agree with you and it's probably better to just explicitly access the `where` field rather than cast such a wide net.